### DISCUSSION TITLE
No new enum-defined component IDs

### DISCUSSION THREAD

**Discussion by plbossart** [thesofproject/sof#7381] [Team Posts]
This topic was mentioned during the TSC but today I reviewed code for the SOF driver that looks like this:

````c
switch (type) {
	case SOF_COMP_EQ_FIR:
	case SOF_COMP_EQ_IIR:
	case SOF_COMP_KEYWORD_DETECT:
	case SOF_COMP_KPB:
	case SOF_COMP_SELECTOR:
		break;
	case SOF_COMP_MUX:
	case SOF_COMP_DEMUX:
		offset = sof_process_load_mux(scomp, process, private);
		break;
	}
````

Clearly this shows it's time to stop adding new components identified with an enum and move to a more elaborate 'type' + UUID identifiers. It's not scalable if we have to update such code every time a new component or 3rd party algorithm is needed.

**Comment by lgirdwood** [comment]
Ack, DAPM only cares about components that change the route, mix or separate paths. We may only need 4 types e.g.
```
MIXER
MUX
DEMUX,
SWITCH
```
DAPM does not care about the others.

**Comment by plbossart** [comment]
we need to add processing as well. Maybe KPB since it's more than just processing.
